server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -378,6 +382,37 @@ public static IndexMergePolicy fromString(String text) {
          */
         new ByteSizeValue(Translog.DEFAULT_HEADER_SIZE_IN_BYTES + 1, ByteSizeUnit.BYTES),
         new ByteSizeValue(Long.MAX_VALUE, ByteSizeUnit.BYTES),
+        new Setting.Validator<ByteSizeValue>() {
+            @Override
+            public void validate(final ByteSizeValue value) {}
+
+            @Override
+            public void validate(ByteSizeValue value, Map<Setting<?>, Object> settings) {
+                boolean isCompositeIndex = (boolean) settings.get(StarTreeIndexSettings.IS_COMPOSITE_INDEX_SETTING);
+                ByteSizeValue compositeIndexMaxFlushThreshold = (ByteSizeValue) settings.get(
+                    COMPOSITE_INDEX_MAX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING
+                );
+                if (isCompositeIndex && value.compareTo(compositeIndexMaxFlushThreshold) > 0) {
+                    throw new IllegalArgumentException(
+                        String.format(
+                            Locale.ROOT,
+                            "You can configure '%s' with upto '%s' for composite index",
+                            INDEX_TRANSLOG_FLUSH_THRESHOLD_SIZE_SETTING.getKey(),
+                            compositeIndexMaxFlushThreshold
+                        )
+                    );
+                }
+            }
```

Comment:
Can you add a test for this?

Reply:
```  
  public void testValidTranslogFlushThresholdSize() {
    public void testInvalidTranslogFlushThresholdSize() {
    public void testUpdateTranslogFlushThresholdSize() {
```

Reply:
Got it. I meant from UT perspective. Lets see if we see issues with coverage and we may have to add UTs as well (I don't think ITs are accounted for in coverage)

Reply:
yeah i agree on that, let me see how to do this